olap/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+mod aggregator;
+mod metrics;
+
+use anyhow::{anyhow, Result};
+use ceramic_metrics::config::Config as MetricsConfig;
+use clap::{Args, Parser, Subcommand, ValueEnum};
+use futures::StreamExt as _;
+use multihash::Multihash;
+use multihash_codetable::Code;
+use multihash_derive::Hasher as _;
+use signal_hook::consts::signal::*;
+use signal_hook_tokio::Signals;
+use tokio::{io::AsyncReadExt as _, sync::oneshot};
+use tracing::{debug, info, warn};
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+enum Command {
+    /// Run a daemon process
+    Daemon(Box<DaemonOpts>),
+}
+
+#[derive(Args, Debug)]
+struct DaemonOpts {
+    /// Bind address of the metrics endpoint.
+    #[arg(
+        short,
+        long,
+        default_value = "127.0.0.1:9464",
```

Comment:
I know this is standard, but wondering if we want a different default if we expect c1 and olap to run on one machine?

Reply:
Good point, its easy to change later and there might be a few things like that to change.